msdsl/expr/format.py

```diff
@@ -207,12 +206,14 @@ def __init__(self, width, min_val=None, max_val=None):
 
     @classmethod
     def width_of(cls, value):
+        # it is better to use bit_length than log2, since log2 has limited
+        # precision, whereas bit_length works on arbitrary integers
         if value < 0:
-            return ceil(log2(0 - value)) + 1
+            return (-1-value).bit_length() + 1

```

Comment:
I didn't know about ``bit_length`` before today, but it looks like it should always be used in place of ``int``/``ceil``/``log2``.  The old way of calculating bit lengths was prone to errors for very large values, due to the limited precision of the floating-point result produced by ``log2``.  There may be some more instances of "old-style" bit length calculations lurking in the code, so we should keep an eye out for those.